test/extensions/filters/http/aws_lambda/aws_lambda_filter_test.cc

```diff
@@ -159,16 +184,16 @@ TEST_F(AwsLambdaFilterTest, PerRouteConfigCorrectClusterMetadata) {
  * then the SigV4 signer will sign with the region where the lambda function is present.
  */
 TEST_F(AwsLambdaFilterTest, PerRouteConfigCorrectRegionForSigning) {
-  setupFilter({arn_, InvocationMode::Synchronous, false /*passthrough*/, ""});
+  setupDownstreamFilter(InvocationMode::Synchronous, false, "");
   const absl::string_view override_region = "us-west-1";
-  const auto per_route_arn =
+  auto route_settings = std::make_shared<FilterSettingsMock>(InvocationMode::Synchronous, true, "");
+  route_settings->arn_ =
       parseArn(fmt::format("arn:aws:lambda:{}:1337:function:fun", override_region)).value();
-  FilterSettings route_settings{per_route_arn, InvocationMode::Synchronous, true /*passthrough*/,
-                                ""};
   ON_CALL(*decoder_callbacks_.route_, mostSpecificPerFilterConfig(_))
-      .WillByDefault(Return(&route_settings));
+      .WillByDefault(Return(&*route_settings));
```

Comment:
Would this work?
`.WillByDefault(Return(route_settings));`

Reply:
probably not, it is a smart pointer, but I could change to .get() it is explicit I need the inner pointer.

Reply:
not needed I can create the concrete object, no need to use smart pointers